backend/icpy/services/chat_service.py

```
@@ -350,6 +350,53 @@ def _derive_session_id_from_file(self, file: Path) -> str:
+    def _surgical_iterate_enabled(self) -> bool:
+        return os.getenv('SURGICAL_ITERATE', '0') in ('1', 'true', 'True')
+
+    def _derive_default_session_name(self, content: str) -> str:
+        try:
+            line = (content or '').strip().splitlines()[0].strip()
+            if not line:
+                return 'New Chat'
+            # Keep it short and filesystem/UI safe
+            line = ' '.join(line.split())
+            return line[:60]
+        except Exception:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)